File: Arrays/4. Sort binary array in linear time/python/code.py
def binSort(arr):
    zeros = 0
    result = [1]*len(arr)
    for elem in arr:
        if (elem == 0):
            zeros += 1

    for i in range(zeros):
        result[i] = 0

    return result

print(binSort(list(map(int,input("Array: ").split(" ")))))

# Input example - 1 0 1 0 1 0 0 1

File: Arrays/4. Sort binary array in linear time/c++/code.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

vector<int> arr;

void printArray(vector<int> &arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        cout << arr[i] << " ";
    }

    cout << endl;
}

void binSort(vector<int> &arr)
{
    vector<int> result;

    for (int i = 0; i < arr.size(); i++)
    {
        if (arr[i] == 0)
        {
            result.push_back(0);
        }
    }

    int size = result.size();

    for (int i = 0; i < arr.size() - size; i++)
    {
        result.push_back(1);
    }

    arr = result;
}

int main()
{
    string input, temp;
    getline(cin, input);
    stringstream ss(input);
    while (getline(ss, temp, ' '))
    {
        arr.push_back(stoi(temp));
    }

    binSort(arr);

    printArray(arr);

    return 0;
}

// Input example - 1 0 1 0 1 0 0 1
